toolchain/check/subst.h

```diff
@@ -32,16 +32,26 @@ class SubstInstCallbacks {
     // Attempt to substitute again on the resulting instruction, acting like
     // recursion on the instruction itself.
     SubstAgain,
+    // Attempt to substitute into the operands of the instruction. If the
+    // operands change, then the rebuilt instruction will be given to `Subst`
+    // again afterward. This allows for the uncommon case of substituting from
+    // the inside out.
+    SubstOperandsAndAgain,
```

Comment:
How would you feel about calling this `SubstOperandsAndSubstAgain` or `SubstOperandsAndRetry` or something like that? "Subst operands and again" is hard for me to understand -- it sounds like we'd be maybe substituting the operands twice.